watertap/tools/oli_api/flash.py

```diff
@@ -808,9 +814,12 @@ def _add_to_json(
         else:
             json_input["params"] = input_data
 
+        output_unit_set_info = {}
+        for k, v in self.output_unit_set.items():
+            output_unit_set_info[k] = v["oli_unit"]
```

Comment:
I turned output_unit_set into a nested FixedKeysDict, which basically means that (1) users can't overwrite keys, which makes sense because these keys correspond to specific OLI output property names, and (2) setting the `oli_unit` or the `pyomo_unit` of a given property will sync with both `oli_unit` and `pyomo_unit` values. See #1388 for more background if that doesn't make sense.